src/promptflow/promptflow/_sdk/_service/apis/run.py

```diff
@@ -50,8 +50,7 @@ def get(self):
             max_results = None
         list_view_type = get_list_view_type(archived_only=archived_only, include_archived=include_archived)
 
-        op = RunOperations()
-        runs = op.list(max_results=max_results, list_view_type=list_view_type)
+        runs = get_client_from_request().runs.list(max_results=max_results, list_view_type=list_view_type)
```

Comment:
Can the value of `get_client_from_request()` be cached and reused? It seems that the same value is being used in many places.

Reply:
Perhaps can add @cache?

Reply:
No.
1. this client is light-weight; instance initialization is not expected to bring perf issue;
2. we create this client to set user-agent for each request, so we will need a lock if we cache a global client;
3. if this really bring perf issue one day, the ideal solution is to provide a way to specify user agent in telemetry for each operation call.

Reply:
#2 means that the clients can be different among requests